ui/src/app/views/resource-card-template.tsx

```diff
@@ -0,0 +1,173 @@
+import {Clickable, MenuItem} from 'app/components/buttons';
+import {ResourceCardBase} from 'app/components/card';
+import {ClrIcon} from 'app/components/icons';
+import {PopupTrigger} from 'app/components/popups';
+import {TextModal} from 'app/components/text-modal';
+import colors from 'app/styles/colors';
+import {reactStyles} from 'app/utils';
+import {navigateAndPreventDefaultIfNoKeysPressed} from 'app/utils/navigation';
+import {ConfirmDeleteModal} from 'app/views/confirm-delete-modal';
+import * as fp from 'lodash';
+import * as React from 'react';
+
+const styles = reactStyles({
+  card: {
+    marginTop: '1rem',
+    justifyContent: 'space-between',
+    marginRight: '1rem',
+    padding: '0.75rem 0.75rem 0rem 0.75rem',
+    boxShadow: '0 0 0 0'
+  },
+  cardName: {
+    fontSize: '18px', fontWeight: 500, lineHeight: '22px', color: colors.accent,
+    cursor: 'pointer', wordBreak: 'break-all', textOverflow: 'ellipsis',
+    overflow: 'hidden', display: '-webkit-box', WebkitLineClamp: 3,
+    WebkitBoxOrient: 'vertical', textDecoration: 'none'
+  },
+  cardDescription: {
+    textOverflow: 'ellipsis', overflow: 'hidden', display: '-webkit-box',
+    WebkitLineClamp: 4, WebkitBoxOrient: 'vertical'
+  },
+  lastModified: {
+    color: colors.primary,
+    fontSize: '11px',
+    display: 'inline-block',
+    lineHeight: '14px',
+    fontWeight: 300,
+    marginBottom: '0.2rem'
+  },
+  resourceType: {
+    height: '22px',
+    width: 'max-content',
+    paddingLeft: '10px',
+    paddingRight: '10px',
+    borderRadius: '4px 4px 0 0',
+    display: 'flex',
+    justifyContent: 'center',
+    color: colors.white,
+    fontFamily: 'Montserrat, sans-serif',
+    fontSize: '12px',
+    fontWeight: 500
+  },
+  cardFooter: {
+    display: 'flex',
+    flexDirection: 'column'
+  }
+});
+
+const defaultProps = {
+  marginTop: '1rem'
+};
+
+export interface Action {
+  displayName: string;
+  onClick: Function;
```

Comment:
Overall: I like the approach of splitting things up! Whatever you can do to decompose the monolith seems sensible to me.

One thing stood out to me, though: down in L135, we're passing in a pretty funky object structure to this onClick handler. It's a slightly surprising pattern since onClick usually takes an event or no argument. I guess the goal here is for concrete "subclasses" to be able to call centrally-defined methods from the "superclass" over here?

On the one hand, it feels a bit like we're rubbing up against React's preference for composition over inheritance by doing inheritance without calling it inheritance... on the other hand, this is already much cleaner than the single component. With a bit of careful commenting & type-safety, this can be totally workable.

But I'd also be interested to see whether a bit of inheritance (as much as React tries to avoid it) might help here, e.g. by allowing the resourceCardFns to be somewhat-awkwardly passed as a function argument...

Reply:
Agreed, the resourceCardFns object is pretty funky. I can look into some react design patterns to see what other people have done.

Reply:
@gjuggler I found another pattern through using higher level components that allows us to reuse state related code. Lmk what you think!

Reply:
I think this looks a lot better! Thanks for following up.